File: ipc_message_queues/receiver.cpp
// Message Queue for Reader Proces
#include <iostream>
#include <sys/ipc.h>
#include <sys/msg.h>
#include <cstring>

struct Message
{
    long messageType;  // Message type
    char payload[100]; // Message content
};

int main()
{
    key_t key = ftok("msgqueue", 65);          // Generate unique key
    int msgid = msgget(key, 0666 | IPC_CREAT); // Get message queue

    Message message;

    msgrcv(msgid, &message, sizeof(message.payload), 1, 0); // Receive message
    std::cout << "Message received: " << message.payload << std::endl;

    msgctl(msgid, IPC_RMID, NULL); // Destroy the message queue

    return 0;
}

File: ipc_message_queues/receiver.py
import sysv_ipc

# Define the same structure as in C++
MESSAGE_TYPE = 1

# Create a unique key for the queue
key = sysv_ipc.ftok("msgqueue", 65)
msg_queue = sysv_ipc.MessageQueue(key)

# Receive a message
message, _ = msg_queue.receive(type=MESSAGE_TYPE)
print("Message received:", message.decode())
